components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/LoginResource.java

```diff
@@ -0,0 +1,124 @@
+import org.json.JSONObject;
+import org.wso2.micro.integrator.management.apis.security.handler.*;
+
+import java.security.NoSuchAlgorithmException;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+
+public class LoginResource implements MiApiResource {
+
+    private static final Log LOG = LogFactory.getLog(LoginResource.class);
+
+    // HTTP method types supported by the resource
+    Set<String> methods;
+
+    public LoginResource() {
+        methods = new HashSet<>();
+        methods.add(Constants.HTTP_POST);
+        methods.add(Constants.HTTP_GET);
+    }
+
+    @Override
+    public Set<String> getMethods() {
+        return methods;
+    }
+
+    @Override
+    public boolean invoke(MessageContext messageContext, org.apache.axis2.context.MessageContext axis2MessageContext,
+                          SynapseConfiguration synapseConfiguration) {
+        //Init token store
+        JWTTokenStore tokenStore = JWTInMemoryTokenStore.
+                getInstance(JWTConfig.getInstance().getJwtConfigDto().getTokenStoreSize());
+
+        //UUID used as unique token
+        UUID uuid = UUID.randomUUID();
+        String randomUUIDString = uuid.toString();
+
+        JWTTokenInfoDTO newToken = new JWTTokenInfoDTO();
+        newToken.setToken(randomUUIDString);
+        newToken.setScope(AuthConstants.JWT_TOKEN_DEFAULT_SCOPE);
+        newToken.setIssuer((String) axis2MessageContext.getProperty(SynapseConstants.SERVER_IP));
+        Long time = new Date().getTime();
+        String expiryConfig = JWTConfig.getInstance().getJwtConfigDto().getExpiry();
+        Long expiryInMins = AuthConstants.DEFAULT_EXPIRY_DURATION;
+        if(expiryConfig != null && !expiryConfig.isEmpty()) {
+            expiryInMins = Long.parseLong(expiryConfig);
+        }
+        newToken.setExpiry(time + (expiryInMins * 60 * 1000));
+        String jwtHash = null;
+        try {
+            jwtHash = populateJWTToken(newToken);
+        } catch (JOSEException e) {
+            LOG.error("Error occurred while generating JWT token", e);
+            handleServerError(axis2MessageContext, "Error occurred while generating JWT token");
+            return true;
+        } catch (NoSuchAlgorithmException e) {
+            LOG.error("Error occurred while generating key pairs", e);
+            handleServerError(axis2MessageContext, "Error occurred while generating key pairs");
+            return true;
+        }
+        newToken.setHash(jwtHash);
+        if(!tokenStore.putToken(jwtHash, newToken)) {
+            handleServerError(axis2MessageContext, "Error occurred while adding token to store");
+            return true;
+        }
+
+        JSONObject jsonPayload = new JSONObject();
+        jsonPayload.put(AuthConstants.RESPONSE_JSON_TOKEN_FIELD, jwtHash);
+        Utils.setJsonPayLoad(axis2MessageContext, jsonPayload);
+        axis2MessageContext.removeProperty(Constants.NO_ENTITY_BODY);
+        return true;
```

Comment:
It seems like the code will anyway return true? Shall we return true only at the end just for clarity?

Reply:
In these code blocks we need to terminate the method execution when an error occurred(After producing an error response). Otherwise we can use a flag whether to produce an error response and check that at the end of the method.